### ISSUE TITLE
Default Configs are not merged in cli-engine

### ISSUE THREAD

**Issue by IanVS** [eslint/eslint#4254]
The [current way](https://github.com/eslint/eslint/blob/6149816042cae4b7082aeb775ba6c899477bd52f/lib/cli-engine.js#L357) that options are merged with a default within the CLIEngine is with:

```js
this.options = assign(Object.create(defaultOptions), options || {});
```

However, this sets `this.options` to an object with the default object as a prototype, and the provided `options` as the object itself.  The defaults themselves are mostly falsy, and the ones which aren't are set to the right defaults within `eslint.js`, so there are no observable bugs because of this.  But, to be correct and merge the properties, it might be better to change the above line to:

```js
this.options = assign({}, defaultOptions, options || {});
```

**Comment by nzakas** [comment]
If there's no bug, why do you think this is incorrect?

**Comment by IanVS** [comment]
Apologies, it was my own misunderstanding here.  I thought there was a problem because:

```js
var options = { rules: { semi: 2 } };

this.options = assign(Object.create(defaultOptions), options || {});

console.log(this.options); // -> { rules: { semi: 2 } }
```

It does not display the default values.  But, I didn't check accessing the properties directly:

```js
console.log(this.options.parser); // -> espree
```

My mistake here, so closing.
